lib/Runtime/Library/JSONStringifier.cpp

```diff
@@ -0,0 +1,817 @@
+        return;
+
+    case TypeIds_UInt64Number:
+        this->SetNumericProperty(static_cast<double>(JavascriptUInt64Number::FromVar(valueObj)->GetValue()), valueObj, prop);
+        return;
+
+#if !FLOATVAR
+    case TypeIds_Number:
+        this->SetNumericProperty(JavascriptNumber::GetValue(valueObj), valueObj, prop);
+        return;
+#endif
+
+    case TypeIds_String:
+        prop->stringValue = JavascriptString::FromVar(valueObj);
+        prop->type = JSONContentType::String;
+        this->totalStringLength = UInt32Math::Add(this->totalStringLength, CalculateStringElementLength(prop->stringValue));
+        return;
+
+    case TypeIds_Symbol:
+        // Treat symbols as undefined
+    case TypeIds_Undefined:
+        // If undefined is in an object, it is not included in the string, and if it is an array element it will show up as "null"
+        // Given that, we can't caluclate the string length here and instead we do it at the usage points
+        prop->type = JSONContentType::Undefined;
+        return;
+
+    default:
+        Assert(UNREACHED);
+        prop->type = JSONContentType::Undefined;
+        return;
+    }
+}
+
+void
+JSONStringifier::ReadData(_In_ Var value, _Out_ JSONProperty* prop)
+{
+    if (RecyclableObject::Is(value))
+    {
+        ReadData(RecyclableObject::FromVar(value), prop);
+        return;
+    }
+
+    // If value isn't a RecyclableObject, it must be a tagged number
+
+    TypeId typeId = JavascriptOperators::GetTypeId(value);
+    switch (typeId)
+    {
+    case TypeIds_Integer:
+    {
+        prop->type = JSONContentType::Number;
+        prop->numericValue.value = value;
+        prop->numericValue.string = this->scriptContext->GetIntegerString(value);
+        this->totalStringLength = UInt32Math::Add(this->totalStringLength, prop->numericValue.string->GetLength());
+        return;
+    }
+#if FLOATVAR
+    case TypeIds_Number:
+        this->SetNumericProperty(JavascriptNumber::GetValue(value), value, prop);
+        return;
+#endif
+    default:
+        Assume(UNREACHED);
+        prop->type = JSONContentType::Undefined;
+        return;
+    }
+}
+
+void
+RecheckValue(_In_ Var value, _Out_ RecyclableObject** valueObj, _Out_ bool* isObject)
+{
+    if (RecyclableObject::Is(value))
+    {
+        *valueObj = RecyclableObject::FromVar(value);
+        *isObject = JavascriptOperators::IsObject(value) != FALSE;
+    }
+    else
+    {
+        *valueObj = nullptr;
+        *isObject = false;
+    }
+}
+
+void
+JSONStringifier::ReadProperty(_In_ JavascriptString* key, _In_opt_ RecyclableObject* holder, _Out_ JSONProperty* prop, _In_opt_ Var value, _In_opt_ const PropertyRecord* propertyRecord, _In_ JSONObjectStack* objectStack)
+{
+    PROBE_STACK(this->scriptContext, Constants::MinStackDefault);
+    if (propertyRecord == nullptr)
+    {
+        propertyRecord = this->GetPropertyRecord(key);
+    }
+    if (value == nullptr)
+    {
+        Assert(holder != nullptr);
```

Comment:
I think this might need to be an `AnalysisAssert` since prefast is complaining on the next line that `holder` might be null.